single_kernel_mongo/utils/helpers.py

```diff
@@ -42,3 +48,21 @@ def hostname_from_shardname(host: str) -> str:
 def unit_number(unit: UnitUpgradePeerData) -> int:
     """Gets the unit number from a unit upgrade peer data."""
     return int(unit.component.name.split("/")[-1])
+
+
+def charm_kind_only(func, charm_kind: CharmKind):
+    """Helpful decorator to ensure the charm kind."""
+
+    def wrapper(self, *args, **kwargs):
+        if self.dependent.name != charm_kind:
+            logger.error(f"{func} called on a non {charm_kind.value} charm. Please fix.")
+            raise InvalidCharmKindError(
+                f"{func} called on a non {charm_kind.value} charm. Please fix."
```

Comment:
isn't this a mistake on the developers side, what should the user do if they see this exception? Sorry if its obvious  🙈

Reply:
sure, but I think: "Unexpected {func} called on a non {charm_kind.value} charm." is good enough :) 

otherwise we set a precedent for when users should/shouldn't contact decs/support

Reply:
You're right